### PR TITLE
SALTO-4019 throw on missing SuiteBundler feature in SuiteApp FileCabinet fetch

### PR DESCRIPTION
**Pull request by dantal4**
Throw an indicative error instead of throwing a generic "Failed to list folders" error.

---

_Additional context for reviewer_

---
_Release Notes_: 
Netsuite Adapter:
- Throw indicative error on missing SuiteBundler feature in SuiteApp FileCabinet fetch

---
_User Notifications_: 
None

### CODE REVIEW

File: packages/netsuite-adapter/src/client/suiteapp_client/suiteapp_client.ts [hunk 1]
```diff
@@ -163,10 +165,17 @@ export default class SuiteAppClient {
...
-        log.error('SuiteQL query error - %s', query, { error })
+        log.warn('SuiteQL query error - %s', query, { error })
         if (error instanceof InvalidSuiteAppCredentialsError) {
           throw error
         }
+        if (axios.isAxiosError(error)) {
+          const errorDetailedMessage = getAxiosErrorDetailedMessage(error)
+          const matchingErrorKey = Object.keys(throwOnErrors).find(e => errorDetailedMessage?.includes(e))
+          if (matchingErrorKey !== undefined) {
+            throw new Error(throwOnErrors[matchingErrorKey])
```
Comment (omrilit) [inline packages/netsuite-adapter/src/client/suiteapp_client/suiteapp_client.ts:176]: Do we want to add a debug log here?

File: packages/netsuite-adapter/src/suiteapp_file_cabinet.ts [hunk 2]
```diff
@@ -159,6 +159,11 @@ const MAX_DEPLOYABLE_FILE_SIZE = 10 * 1024 * 1024
 const DEPLOY_CHUNK_SIZE = 50
 const MAX_ITEMS_IN_WHERE_QUERY = 200
 
+export const THROW_ON_MISSING_FEATURE_ERROR = {
```
Comment (omrilit) [inline packages/netsuite-adapter/src/suiteapp_file_cabinet.ts:162]: nit: 
```suggestion
export const THROW_ON_MISSING_FEATURE_ERROR : Record<string, string> = {
```

**Review by omrilit — APPROVED** [review]
Thanks!

File: packages/netsuite-adapter/src/client/suiteapp_client/suiteapp_client.ts (hunk 1 above)
Comment (dantal4) [reply packages/netsuite-adapter/src/client/suiteapp_client/suiteapp_client.ts:176]: It is thrown and fails the fetch operation, so it feels redundant, isn't it?

**Pull request merged**